### ISSUE TITLE
linux build issues

### ISSUE THREAD

**Issue by RoySullivan** [ivansafrin/Polycode#712]
how to fix this? it's from sanghakchun/Polycode branch linux-build-fix

../../src/modules/physics2D/PolyPhysicsScreen.cpp: In constructor ‘Polycode::PhysicsScene2D::PhysicsScene2D()’:
../../src/modules/physics2D/PolyPhysicsScreen.cpp:198:57: error: invalid conversion from ‘int’ to ‘Polycode::Core*’ [-fpermissive]
 PhysicsScene2D::PhysicsScene2D() : Scene(Scene::SCENE_2D) {
                                                         ^
In file included from ../../include/polycode/modules/physics2D/PolyPhysicsScreen.h:26:0,
                 from ../../src/modules/physics2D/PolyPhysicsScreen.cpp:24:
../../include/polycode/core/PolyScene.h:61:3: note:   initializing argument 1 of ‘Polycode::Scene::Scene(Polycode::Core*)’
   Scene(Core *core);
   ^
../../src/modules/physics2D/PolyPhysicsScreen.cpp: In constructor ‘Polycode::PhysicsScene2D::PhysicsScene2D(Number, int, int)’:
../../src/modules/physics2D/PolyPhysicsScreen.cpp:202:111: error: invalid conversion from ‘int’ to ‘Polycode::Core*’ [-fpermissive]
 ber worldScale, int velIterations, int posIterations): Scene(Scene::SCENE_2D) {
                                                                             ^
In file included from ../../include/polycode/modules/physics2D/PolyPhysicsScreen.h:26:0,
                 from ../../src/modules/physics2D/PolyPhysicsScreen.cpp:24:
../../include/polycode/core/PolyScene.h:61:3: note:   initializing argument 1 of ‘Polycode::Scene::Scene(Polycode::Core*)’
   Scene(Core *core);
   ^
Makefile:55: recipe for target '../../src/modules/physics2D/PolyPhysicsScreen.o' failed
make: *** [../../src/modules/physics2D/PolyPhysicsScreen.o] Error 1

**Comment by zeebok** [comment]
Is this a straight clone of his repo? I can't vouch for that, but if you pull Ivan's repo, checkout the goodbye_cmake branch, and then pull in sanghakchun's merge request that should work. That's how I got myself up and running.

**Comment by RoySullivan** [comment]
@zeebok i forked goodbye_cmake and pull merge with sanghakchun linux-build-fix branch, still giving me the same previous errors like missing files that i have to copy from the master branch, i also had to apply this https://github.com/fodinabor/Polycode/commit/e59fbe395d7c8061e50010d8fb9259039ee0c7b7

but other problems emerge like this:
from ../../src/modules/physics3D/PolyPhysicsScene.cpp:27:
../../include/polycode/modules/physics3D/PolyCollisionScene.h:90:76: error: ISO C++ forbids declaration of ‘PhysicsScene’ with no type [-fpermissive]
   PhysicsScene(Core* core, int maxSubSteps = 0, Vector3 size = Vector3(200)); 

am i doing something wrong?

**Comment by zeebok** [comment]
Ah I haven't applied fodinabor's patch yet, let me take a look.

**Comment by zeebok** [comment]
I am assuming you're trying to compile using `make all` ? Using fodinabor's branch and the linux build fix, Physics3D should build. I am getting errors for Physics2D but I don't think that was suppose to be fixed on fodinabor's branch so I am not surprised by that.

You should be able to just download fodi's branch and then apply the linux build fix; it seems to be up-to-date with Ivan's branch.

**Comment by RoySullivan** [comment]
yeah doing that compiles Physics3D, so does sanghakchun/Polycode nuevo branch, but still the same problems with Physics2D persist.

**Comment by fodinabor** [comment]
I think the Physics2D is not updated yet...

**Comment by RoySullivan** [comment]
@fodinabor yeah, should i close this issues, since it's technically not an issue?

**Comment by fodinabor** [comment]
well the issue is rather that Physics2D is not up-to-date - if you wish you can create a new issue for this and close this one..

**Comment by ghost** [comment]
I just cant understand why someone with experience in polycode cant make a pull request with working build system.

**Comment by fodinabor** [comment]
The build system (system specific) should be working? What do you mean?

**Comment by zeebok** [comment]
I think he means the pull request for the linux build not being able to fix/compile all of the modules as well as fixing the core framework

**Comment by ghost** [comment]
@zeebok , exactly. I thought I was clear enough.
